wemake_python_styleguide/visitors/ast/compares.py

```diff
@@ -237,3 +238,24 @@ def _is_float_or_complex(self, node: ast.AST) -> bool:
             node.value,
             float | complex,
         )
+
+
+@final
+class NotInUnaryVisitor(BaseNodeVisitor):
+    """Forbids ``not a in b`` which should be written as ``a not in b``."""
+
+    def visit_UnaryOp(self, node: ast.UnaryOp) -> None:
+        """Detects ``not (a in b)`` pattern syntactically.
+
+        Python parses both ``not a in b`` and ``not (a in b)`` as a
+        ``UnaryOp(Not, Compare(left, In, right))``; we catch this shape
+        and suggest ``a not in b`` instead.
+        """
+        if isinstance(node.op, ast.Not) and isinstance(
+            node.operand,
+            ast.Compare,
+        ):
+            comp = node.operand
+            if len(comp.ops) == 1 and isinstance(comp.ops[0], ast.In):
+                self.add_violation(NotInWithUnaryOpViolation(node))
```

Comment:
```suggestion
   def _check_legacy_not_int(self, node: ast.UnaryOp) -> None:
        if isinstance(node.op, ast.Not) and isinstance(
            node.operand,
            ast.Compare,
        ):
            comp = node.operand
            if len(comp.ops) == 1 and isinstance(comp.ops[0], ast.In):
                self.add_violation(NotInWithUnaryOpViolation(node))
```

and call this method in `visit_UnaryOp`